packages/stable/src/api/records/recordsApi.ts

```diff
@@ -79,4 +83,54 @@ export class RecordsAPI extends BaseResourceAPI<RecordItem> {
     });
     return this.addToMapAndReturn(response.data.items, response);
   };
+
+  /**
+   * [Sync records from a stream](https://developer.cognite.com/api#tag/Records/operation/syncRecords)
+   *
+   * Sync records from a stream using a cursor-based approach. Supports auto-pagination.
+   *
+   * ```js
+   * // Get first page
+   * const response = await client.records.sync('my_stream', {
+   *   initializeCursor: '1d-ago',
+   *   sources: [{ source: { type: 'container', space: 'mySpace', externalId: 'myContainer' }, properties: ['*'] }],
+   * });
+   *
+   * // Auto-paginate to array
+   * const allRecords = await client.records
+   *   .sync('my_stream', { initializeCursor: '1d-ago', sources: [{ source: { type: 'container', space: 'mySpace', externalId: 'myContainer' }, properties: ['*'] }] })
+   *   .autoPagingToArray({ limit: 10000 });
+   *
+   * // Iterate with for-await
+   * for await (const record of client.records.sync('my_stream', { initializeCursor: '1d-ago' })) {
+   *   console.log(record);
+   * }
+   * ```
+   */
+  public sync = (
+    streamExternalId: string,
+    request: RecordSyncRequest
+  ): CursorAndAsyncIterator<SyncRecordItem> => {
+    const path = this.url(`${streamExternalId}/records/sync`);
+
+    const callSyncEndpoint = async (params?: RecordSyncRequest) => {
+      const response = await this.post<RecordSyncResponse>(path, {
+        data: params,
+      });
+      const items = this.addToMapAndReturn(
+        response.data.items,
+        response
+      ) as SyncRecordItem[];
```

Comment:
`as` is scary in Typescript as we disable type-checking.

Just do:

```suggestion
      );
```

Reply:
Addressed in https://github.com/cognitedata/cognite-sdk-js/pull/1344/commits/5ca23fbb2d50b008e473401bb60eaffdb4d01c9d